src/Go/boardAnalysis/goAI.ts

```diff
@@ -21,13 +21,23 @@ import { findAnyMatchedPatterns } from "./patternMatching";
 import { WHRNG } from "../../Casino/RNG";
 import { Go, GoEvents } from "../Go";
 
+let isAIBusy = false;
+
 /**
  * Retrieves a move from the current faction in response to the player's move
  */
 export async function getAIMove(boardState: BoardState): Promise<Play> {
-  let resolve: (value: Play) => void;
+  // If an AI move is already in progress, return the already in progress move
+  if (isAIBusy && Go.nextTurn) return Go.nextTurn;
+  // Disallow new AI moves to be triggered during this one
+  isAIBusy = true;
+  let resolve: (play: Play) => void;
   Go.nextTurn = new Promise<Play>((res) => {
-    resolve = res;
+    resolve = (play) => {
+      // Once the promise resolves, stop blocking new AI moves
+      isAIBusy = false;
```

Comment:
Subtle point: I think you should set this in a `finally` chained off of `getMove`. I don't *think* it makes a difference now, but it would if any of these promises rejected.

Reply:
Good idea, implemented in a471108 / 24cd914